helper/common/common.go

```diff
@@ -99,22 +102,137 @@ func DirectoryExists(directoryPath string) bool {
 	return true
 }
 
-// createDir creates a file system directory if it doesn't exist
-func createDir(path string) error {
-	_, err := os.Stat(path)
+// Checks if the file at the specified path exists
+func FileExists(filePath string) bool {
+	// Grab the absolute filepath
+	pathAbs, err := filepath.Abs(filePath)
+	if err != nil {
+		return false
+	}
+
+	// Check if the file exists, and that it's actually a file if there is a hit
+	if fileInfo, statErr := os.Stat(pathAbs); os.IsNotExist(statErr) || (fileInfo != nil && fileInfo.IsDir()) {
+		return false
+	}
+
+	return true
+}
+
+// Creates a directory at path and with perms level permissions.
+// If directory already exists, owner and permissions are verified.
+func CreateDirSafe(path string, perms fs.FileMode) error {
+	info, err := os.Stat(path)
+	// check if an error occurred other than path not exists
 	if err != nil && !os.IsNotExist(err) {
 		return err
 	}
 
-	if os.IsNotExist(err) {
-		if err := os.MkdirAll(path, os.ModePerm); err != nil {
+	// create directory if it does not exist
+	if !DirectoryExists(path) {
+		if err := os.MkdirAll(path, perms); err != nil {
 			return err
 		}
+
+		return nil
+	}
+
+	// verify that existing directory's owner and permissions are safe
+	err = verifyFileAndTryUpdatePermissions(path, info, perms)
+	if err != nil {
+		return err
 	}
 
 	return nil
 }
 
+// Creates a file at path and with perms level permissions.
+// If file already exists, owner and permissions are verified.
+// If shouldNotExist is true, an error is returned if file exists
+func CreateFileSafe(path string, data []byte, perms fs.FileMode, shouldNotExist bool) error {
+	info, err := os.Stat(path)
+	// check if an error occurred other than path not exists
+	if err != nil && !os.IsNotExist(err) {
+		return err
+	}
+
+	fileExists := FileExists(path)
+	if fileExists && shouldNotExist {
+		return fmt.Errorf("%s already initialized", path)
+	}
```

Comment:
I would remove this logic and call it where its needed. Most frequent usage of third param for this kind of functions is also bool but should deal with the logic if the file should be created if it doesn't exist. If you still feel like you need a function that does both, maybe its better idea to create a wrapper function instead, something along the line checkOrCreate.

Reply:
done